File: library/src/main/java/com/tundem/aboutlibraries/ui/LibsActivity.java
package com.tundem.aboutlibraries.ui;

import android.app.ActionBar;
import android.graphics.Color;
import android.graphics.Rect;
import android.os.Bundle;
import android.support.v4.app.FragmentActivity;
import android.support.v4.app.FragmentManager;
import android.text.TextUtils;
import android.view.MenuItem;
import android.widget.FrameLayout;

import com.tundem.aboutlibraries.Libs;
import com.tundem.aboutlibraries.R;
import com.tundem.aboutlibraries.ui.utils.UIUtils;
import com.tundem.aboutlibraries.ui.view.DrawInsetsFrameLayout;

/**
 * Created by mikepenz on 04.06.14.
 */
public class LibsActivity extends FragmentActivity {

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        boolean customTheme = false;
        Bundle bundle = getIntent().getExtras();
        if (bundle != null) {
            int themeId = bundle.getInt(Libs.BUNDLE_THEME, -1);
            if (themeId != -1) {
                customTheme = true;
                setTheme(themeId);
            }
        }

        setContentView(R.layout.activity_opensource);

        boolean usedAccentColor = false;
        if (bundle != null) {
            String accentColorString = bundle.getString(Libs.BUNDLE_ACCENTCOLOR, "");
            boolean useTranslucentDecor = bundle.getBoolean(Libs.BUNDLE_TRANSLUCENTDECOR, false);

            if (!TextUtils.isEmpty(accentColorString)) {
                usedAccentColor = true;

                int accentColor = Color.parseColor(accentColorString);
                int accentSecondaryColor = Color.parseColor("#88" + Integer.toHexString(accentColor).toUpperCase().substring(2));
                int backgroundColor = Color.parseColor("#e5e5e5");

                UIUtils.init(getApplicationContext(), accentColor, accentSecondaryColor, backgroundColor, true, true, true, true);

                if (useTranslucentDecor) {
                    UIUtils.getInstance().initActivity(this);
                }

                DrawInsetsFrameLayout drawInsetsFrameLayout = (DrawInsetsFrameLayout) findViewById(R.id.drawinsetsframelayout);
                drawInsetsFrameLayout.setOnInsetsCallback(new DrawInsetsFrameLayout.OnInsetsCallback() {
                    @Override
                    public void onInsetsChanged(Rect insets) {
                        findViewById(R.id.frame_container).setLayoutParams(UIUtils.getInstance().handleTranslucentDecorMargins(((FrameLayout.LayoutParams) findViewById(R.id.frame_container).getLayoutParams()), insets));
                    }
                });
            }
        }


        LibsFragment fragment = new LibsFragment();
        fragment.setArguments(bundle);

        ActionBar ab = getActionBar();
        if (ab != null) {
            if (!customTheme && usedAccentColor) {
                ab.setBackgroundDrawable(null);
            }
            ab.setDisplayHomeAsUpEnabled(true);
            ab.setDisplayShowTitleEnabled(false);
            ab.setDisplayUseLogoEnabled(true);
        }

        FragmentManager fragmentManager = getSupportFragmentManager();
        fragmentManager.beginTransaction().replace(R.id.frame_container, fragment).commit();
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home: {
                LibsActivity.this.finish();
                return true;
            }
            default:
                return false;
        }
    }
}

File: plugin-build/library/src/main/java/com/mikepenz/aboutlibraries/ui/LibsActivity.kt
package com.mikepenz.aboutlibraries.ui

import android.os.Bundle
import android.view.Gravity
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import com.mikepenz.aboutlibraries.Libs
import com.mikepenz.aboutlibraries.Libs.Companion.BUNDLE_EDGE_TO_EDGE
import com.mikepenz.aboutlibraries.R
import com.mikepenz.aboutlibraries.util.applyEdgeSystemUi
import com.mikepenz.aboutlibraries.util.doOnApplySystemWindowInsets


/**
 * Created by mikepenz on 04.06.14.
 */
open class LibsActivity : AppCompatActivity() {

    public override fun onCreate(savedInstanceState: Bundle?) {
        //set the theme
        var edgeToEdge = false
        val bundle = intent.extras
        if (bundle != null) {
            edgeToEdge = bundle.getBoolean(BUNDLE_EDGE_TO_EDGE)
            if (edgeToEdge) {
                applyEdgeSystemUi()
            }
        }
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_opensource)
        var title = ""
        if (bundle != null) {
            title = bundle.getString(Libs.BUNDLE_TITLE, "")
        }
        val fragment = LibsSupportFragment()
        fragment.arguments = bundle

        // Handle Toolbar
        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)

        // Support ActionBar :D
        val ab = supportActionBar
        if (ab != null) {
            // SetUp ActionBar
            ab.setDisplayHomeAsUpEnabled(true)
            ab.setDisplayShowTitleEnabled(title.isNotEmpty())
            ab.title = title
        }

        // apply insets
        toolbar.doOnApplySystemWindowInsets(Gravity.TOP, Gravity.START, Gravity.END)

        supportFragmentManager.beginTransaction().replace(R.id.frame_container, fragment).commit()
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            android.R.id.home -> {
                this@LibsActivity.finish()
                true
            }
            else -> false
        }
    }
}
